src/client.rs

```diff
@@ -235,7 +255,7 @@ impl Client {
     /// If room_id is not one of the joined rooms or couldn't retrieve it.
     ///
     pub async fn send_message(&self, room_id: &RoomId, message: String) {
-        let client = self.inner.lock().await;
+        let client = &self.inner;
```

Comment:
Is the reference `&` still needed if `self` is already used as a shared reference here (also applies to previous changes as well)?

Reply:
it is. the reason is that `self.inner` doesn't implement the `Copy` trait and the compiler complains about this move.

I think I can remove the `Arc` and use it as it is (without local vars)